File: src/tensor/unittest/forward_test/test_forward.cc
#include <iostream>

#include "tensor.h"

int main() {
  auto a = Tensor(Shape{2, 2}, std::vector<float>{1.0, 2.0, 3.0, 4.0});
  auto b = Tensor(Shape{2}, std::vector<float>{1.0, 2.0});

  auto c = a + b;

  c.dump_tensor(std::cout);

  return 0;
}

File: src/tensor/unittest/forward_test/test_forward.py
import torch

def dump_torch_tensor(t):
    print(t.ndim)
    print(" ".join(map(str, t.shape)))
    for val in t.flatten():
        print(f"{val.item():.6f}", end=" ")
    print()

if __name__ == "__main__":
    a = torch.tensor([[1.0, 2.0], [3.0, 4.0]], dtype=torch.float32)
    b = torch.tensor([1.0, 2.0], dtype=torch.float32)
    c = a + b
    dump_torch_tensor(c)